backend/src/main/java/pt/ulisboa/tecnico/socialsoftware/tutor/question/domain/Question.java

```diff
@@ -19,50 +19,55 @@
 
 @Entity
 @Table(name = "questions")
-public class Question implements DomainEntity {
+@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
+@DiscriminatorColumn(name = "question_type",
+        columnDefinition = "smallint not null default 0",
+        discriminatorType = DiscriminatorType.INTEGER)
+public abstract class Question implements DomainEntity {
+    public static class QuestionTypes{
+        public static final String MultipleChoice = "0";
+    }
+
     public enum Status {
         DISABLED, REMOVED, AVAILABLE
     }
 
     @Id
     @GeneratedValue(strategy = GenerationType.IDENTITY)
-    private Integer id;
+    protected Integer id;
```

Comment:
Why protected? Avoid doing this since we are self-encapsulated field and all accesses to the domain attributes should be done through the accessors